packages/fhir-converter/src/lambdas/ccda-to-fhir.js

```diff
@@ -0,0 +1,192 @@
+var path = require("path");
+var fs = require("fs");
+var Promise = require("promise");
+var compileCache = require("memory-cache");
+var constants = require("./lib/constants/constants");
+var errorCodes = require("./lib/error/error").errorCodes;
+var errorMessage = require("./lib/error/error").errorMessage;
+var HandlebarsConverter = require("./lib/handlebars-converter/handlebars-converter");
+var dataHandlerFactory = require("./lib/dataHandler/dataHandlerFactory");
+var {
+  extractEncounterTimePeriod,
+  getEncompassingEncounterId,
+} = require("./lib/inputProcessor/dateProcessor");
+
+const { createNamespace } = require("cls-hooked");
+var session = createNamespace(constants.CLS_NAMESPACE);
+
+var rebuildCache = true;
+
+function GetHandlebarsInstance(dataTypeHandler, templatesMap) {
+  // New instance should be created when using templatesMap
+  let needToUseMap =
+    templatesMap && Object.entries(templatesMap).length > 0 && templatesMap.constructor === Object;
+  var instance = HandlebarsConverter.instance(
+    needToUseMap ? true : rebuildCache,
+    dataTypeHandler,
+    path.join(constants.BASE_TEMPLATE_FILES_LOCATION, dataTypeHandler.dataType),
+    templatesMap
+  );
+  rebuildCache = needToUseMap ? true : false; // New instance should be created also after templatesMap usage
+
+  return instance;
+}
+
+function generateResult(
+  dataTypeHandler,
+  dataContext,
+  template,
+  patientId,
+  encounterTimePeriod,
+  encompassingEncounterId
+) {
+  var result = dataTypeHandler.postProcessResult(
+    template(dataContext, {
+      data: { metriportPatientId: patientId, encounterTimePeriod, encompassingEncounterId },
+    })
+  );
+  return Object.assign(dataTypeHandler.getConversionResultMetadata(dataContext.msg), {
+    fhirResource: result,
+  });
+}
+
+async function ccdaToFhir(ccda, patientId) {
+  return new Promise((fulfill, reject) => {
+    session.run(() => {
+      let srcData = ccda;
+      if (!srcData || srcData.length == 0) {
+        reject({
+          status: 400,
+          resultMsg: "No srcData provided.",
+        });
```

Comment:
We should separate the processing/converting from the HTTP response details. I'd suggest using the approach we've been using w/ custom Error classes.

Reply:
ditto - trying to keep the logic as similar as possible ahead of a refactor for now, captured this in the follow on ticket